monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -74,3 +90,29 @@ def _scan_tcp_ports(self, ip: str, ports: List[int], timeout: float, stop: Event
             port_scan_data[p] = self._puppet.scan_tcp_port(ip, p, timeout)
 
         return port_scan_data
+
+    @staticmethod
+    def _found_open_port(port_scan_data: Dict[int, PortScanData]):
+        for psd in port_scan_data.values():
+            if psd.status == PortStatus.OPEN:
```

Comment:
Can we `pds.has_open_port()` to not expose the internal structure?

Reply:
Ahhh, the `PortScanData` is a named tuple... Not sure we want to convert to a dataclass, but worth considering

Reply:
I'm not sure we get any benefit from converting it. Type hinting, but the types are all fairly simple. It's a toss up for me.